tests/cupyx_tests/scipy_tests/sparse_tests/test_index.py

```diff
@@ -9,17 +9,258 @@
     pass
 
 import cupy
+import cupyx
 from cupy import testing
 from cupyx.scipy import sparse
 
 
+def _get_index_combos(idx):
+    return [dict['arr_fn'](idx, dtype=dict['dtype'])
+            for dict in testing.product({
+                "arr_fn": [numpy.array, cupy.array],
+                "dtype": [numpy.int32, numpy.int64]
+            })]
+
+
 def _check_shares_memory(xp, sp, x, y):
     if sp.issparse(x) and sp.issparse(y):
         assert not xp.shares_memory(x.indptr, y.indptr)
         assert not xp.shares_memory(x.indices, y.indices)
         assert not xp.shares_memory(x.data, y.data)
 
 
+@testing.parameterize(*testing.product({
+    'format': ['csr', 'csc'],
+    'density': [0.9],
+    'dtype': ['float32', 'float64', 'complex64', 'complex128'],
+    'n_rows': [25, 150],
+    'n_cols': [25, 150]
+}))
+@testing.with_requires('scipy>=1.4.0')
+@testing.gpu
+class TestSetitemIndexing(unittest.TestCase):
```

Comment:
Could you add tests for the case of
- Boolean mask of numpy ndarray
- numpy/cupy ndarray `data` which is set to sparse array
- numpy scalar and cupy 0-dim `data` which is set to sparse array

Reply:
Please add numpy scalar and cupy 0-dim data cases.